src/rules.ts

```diff
@@ -186,7 +186,9 @@ function evaluateCondition(subjectAttributes: Record<string, any>, condition: Co
         }
         return compareNumber(value, condition.value, (a, b) => a < b);
       case OperatorType.MATCHES:
-        return new RegExp(condition.value as string).test(value as string);
+        return new RegExp(`^${condition.value as string}$`).test(value as string);
+      case OperatorType.NOT_MATCHES:
+        return !new RegExp(`^${condition.value as string}$`).test(value as string);
```

Comment:
Note: the not_matches implementation was missing previously

Reply:
good catch!